### ISSUE TITLE
registerStreamProtocol does not work with XHR or Fetch

### ISSUE THREAD

**Issue by kewde** [electron/electron#13623]
* Electron Version: 1.8.4 & 1.8.7
* Operating System (Platform and Version): Linux, Debian
* Last known working Electron version: /

**Expected Behavior**
An XHR or Fetch request to a custom stream protocol (registerStreamProtocol) should return the stream data.

**Actual behavior**
The request remains permanently pending, it never receives any data.

**To Reproduce**
I made branch to test it out, I have 4 examples there with documentation.
```sh
$ git clone https://github.com/kewde/electron-sandbox-boilerplate.git -b bug-stream-protocol
$ cd electron-sandbox-boilerplate/sandbox-stream-protocol
$ yarn install
$ yarn run start
```
**Screenshots**

**Additional Information**
Additional information in the source code.

tl;dr:
Tested 4 methods:
* XHR + registerBufferProtocol = works :+1: 
* XHR + registerStreamProtocol = :-1:
* Fetch + registerStreamProtocol = :-1: 
* Fetch + interceptStreamProtocol('http',... = :-1:

**Comment by kewde** [comment]
Potentially the same bug #13519

**Comment by ccorcos** [comment]
I'm using electron 3, but I noticed that you need `protocol.registerStreamProtocol` for XHR to work and you also need `webFrame.registerURLSchemeAsPrivileged` for fetch to work.

https://github.com/ccorcos/electron-protocol-example

**Comment by sofianguy** [comment]
Thank you for taking the time to report this issue and helping to make Electron better.

The version of Electron you reported this on has been superseded by newer releases. 

If you're still experiencing this issue in Electron v2.0.x or later, please add a comment specifying the version you're testing with and any other new information that a maintainer trying to reproduce the issue should know.

I'm setting the `blocked/need-info` label for the above reasons.

Thanks in advance! Your help is appreciated.

**Comment by sofianguy** [comment]
Thank you for your issue!

We haven't gotten a response to our questions in our comment above. With only the information that is currently in the issue, we don't have enough information to take action. I'm going to close this but don't hesitate to reach out if you have or find the answers we need, we'll be happy to reopen the issue.
